File: 8kyu/(8 kyu) DNA to RNA Conversion/(8 kyu) DNA to RNA Conversion.java
// #1
public class Bio {
  public String dnaToRna(String dna) {
    String res = "";
    for (int i = 0; i < dna.length(); i++) {
      char c = dna.charAt(i);
      if (c == 'T') {
        res += "U";
      } else {
        res += c;
      }
    }
    return res;
  } 
}

// #2
public class Bio {
  public String dnaToRna(String dna) {
    String res = "";
    for (char c: dna.toCharArray()) {
      res += (c == 'T') ? "U" : c;
    }
    return res;
  } 
}

// #3
public class Bio {
  public String dnaToRna(String dna) {
    return dna.replaceAll("T", "U");
  } 
}

File: 8kyu/(8 kyu) DNA to RNA Conversion/(8 kyu) DNA to RNA Conversion.dart
// 1 Plain solution
String rnaToDna(String dna) {
  String res = '';
  for (int i = 0; i < dna.length; i++) {
    String c = dna.substring(i, i + 1);
    if (c == 'T') {
      res += 'U';
    } else {
      res += c;
    }
  }
  return res;
}

// 2 Optimized solution
String rnaToDna(String dna) {
  String res = '';
  for (String c in dna.split('')) {
    res += (c == 'T') ? 'U' : c;
  }
  return res;
}

// 3 Clever solution
String rnaToDna(String dna) {
  return dna.replaceAll('T', 'U');
}

// 4 Coding golf
String rnaToDna(String dna) => dna.replaceAll('T', 'U');
